From 073986abc5463e3333f5576eda267e4c2f3c1460 Mon Sep 17 00:00:00 2001
From: Quentin Guilloteau <Quentin.Guilloteau@inria.fr>
Date: Wed, 19 Apr 2023 16:24:22 +0200
Subject: [PATCH 3/3] events for get /jobs

---
 oar/rest_api/views/job.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/oar/rest_api/views/job.py b/oar/rest_api/views/job.py
index 4a15eda..7c426d0 100644
--- a/oar/rest_api/views/job.py
+++ b/oar/rest_api/views/job.py
@@ -9,6 +9,7 @@ from oar.cli.oarhold import oarhold
 from oar.cli.oarresume import oarresume
 from oar.lib import Job, db
 from oar.lib.submission import JobParameters, Submission, check_reservation
+from oar.lib.event import get_job_events
 
 from ..utils import Arg
 from . import Blueprint
@@ -324,6 +325,7 @@ def show(job_id, detailed=None):
         job_resources = db.queries.get_assigned_jobs_resources([job])
         attach_resources(g.data, job_resources)
         attach_nodes(g.data, job_resources)
+    g.data["events"] = get_job_events(job_id)
     attach_links(g.data)
 
 
-- 
2.38.1

