From 1ef769ebf3ac1755606f986d79d8af6c967d0f38 Mon Sep 17 00:00:00 2001
From: Quentin Guilloteau <Quentin.Guilloteau@inria.fr>
Date: Sun, 30 Apr 2023 00:05:43 +0200
Subject: [PATCH 4/4] resubmit job id

---
 oar/lib/job_handling.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/oar/lib/job_handling.py b/oar/lib/job_handling.py
index 091fa83..3753b44 100644
--- a/oar/lib/job_handling.py
+++ b/oar/lib/job_handling.py
@@ -984,7 +984,7 @@ def resubmit_job(job_id):
         job.start_time = 0
     job.message = ""
     job.scheduler_info = ""
-    job.exit_code = None
+    # job.exit_code = None
     job.assigned_moldable_job = 0
 
     db.session.add(job)
@@ -2020,7 +2020,7 @@ def check_end_of_job(
                 "idempotent" in job_types.keys()
             ):
                 if is_an_event_exists(job_id, "BESTEFFORT_KILL"):
-                    new_job_id = resubmit_job(job_id)
+                    (_, new_job_id) = resubmit_job(job_id)
                     logger.warning(
                         "We resubmit the job "
                         + str(job_id)
-- 
2.38.1

