diff --git a/dummy_components/Monitor/RegaleMonitorServer_impl.c b/dummy_components/Monitor/RegaleMonitorServer_impl.c
index 048ee2c..311aa15 100644
--- a/dummy_components/Monitor/RegaleMonitorServer_impl.c
+++ b/dummy_components/Monitor/RegaleMonitorServer_impl.c
@@ -1,4 +1,5 @@
-#include <regale_monitor_types.h>
+#include <regale_monitor_server_impl.h>
+
 regale_state_t regale_report_node_telemetry_server(regale_node_id_t *node,regale_node_data_t *data)
 {
 	regale_verbose_node_data(0, node, data);
diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index d45b687..2d0714a 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -1,5 +1,5 @@
-set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+# set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+# set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
 file(GLOB SOURCE_FILES *.c *.cpp)
 
diff --git a/examples/main_publisher.c b/examples/main_publisher.c
index c6e7d42..080c248 100644
--- a/examples/main_publisher.c
+++ b/examples/main_publisher.c
@@ -1,4 +1,4 @@
-#include "regale.h"
+#include "regale_core.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
diff --git a/examples/main_publisher_02.c b/examples/main_publisher_02.c
index f505740..438e6f8 100644
--- a/examples/main_publisher_02.c
+++ b/examples/main_publisher_02.c
@@ -1,4 +1,4 @@
-#include "regale.h"
+#include "regale_core.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
diff --git a/examples/main_publisher_02_cpp.cpp b/examples/main_publisher_02_cpp.cpp
index f505740..438e6f8 100644
--- a/examples/main_publisher_02_cpp.cpp
+++ b/examples/main_publisher_02_cpp.cpp
@@ -1,4 +1,4 @@
-#include "regale.h"
+#include "regale_core.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
diff --git a/examples/main_publisher_cpp.cpp b/examples/main_publisher_cpp.cpp
index c6e7d42..080c248 100644
--- a/examples/main_publisher_cpp.cpp
+++ b/examples/main_publisher_cpp.cpp
@@ -1,4 +1,4 @@
-#include "regale.h"
+#include "regale_core.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
diff --git a/examples/main_subscriber.c b/examples/main_subscriber.c
index 1bc65bd..e5f7336 100644
--- a/examples/main_subscriber.c
+++ b/examples/main_subscriber.c
@@ -1,4 +1,4 @@
-#include "regale.h"
+#include "regale_core.h"
 #include <stdio.h>
 #include <unistd.h>
 #include <string.h>
diff --git a/examples/main_subscriber_02.c b/examples/main_subscriber_02.c
index 5975cfc..2628004 100644
--- a/examples/main_subscriber_02.c
+++ b/examples/main_subscriber_02.c
@@ -1,4 +1,4 @@
-#include "regale.h"
+#include "regale_core.h"
 #include <stdio.h>
 #include <unistd.h>
 #include <string.h>
diff --git a/examples/main_subscriber_02_cpp.cpp b/examples/main_subscriber_02_cpp.cpp
index 5975cfc..2628004 100644
--- a/examples/main_subscriber_02_cpp.cpp
+++ b/examples/main_subscriber_02_cpp.cpp
@@ -1,4 +1,4 @@
-#include "regale.h"
+#include "regale_core.h"
 #include <stdio.h>
 #include <unistd.h>
 #include <string.h>
diff --git a/src/regale.cpp b/src/regale.cpp
index 6144e7d..e398834 100644
--- a/src/regale.cpp
+++ b/src/regale.cpp
@@ -1,4 +1,4 @@
-#include "regale.h"
+#include "regale_core.h"
 #include <string.h>
 #include "RegaleSubscriber.h"
 #include "RegalePublisher.h"
