diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5392127c5d..d25a25e5f8 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1047,7 +1047,7 @@ endif()
 target_include_directories(libscip BEFORE PUBLIC
     $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-    $<INSTALL_INTERFACE:include>)
+    $<INSTALL_INTERFACE:${CMAKE_CURRENT_INCLUDE_DIR}>)
 
 target_link_libraries(libscip
     PRIVATE
@@ -1126,7 +1126,7 @@ set_target_properties(libscip PROPERTIES
 target_include_directories(scip BEFORE PUBLIC
     $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-    $<INSTALL_INTERFACE:include>)
+    $<INSTALL_INTERFACE:${CMAKE_CURRENT_INCLUDE_DIR}>)
 
 # set the install rpath to the installed destination
 set_target_properties(scip PROPERTIES
